### PR TITLE
Add support for man pages

### LINKED ISSUE #171: Man pages (improvement)
There are no man pages for waybar. I think this would be useful for users that might need to quickly look up a config option. More or less just manualizing the wiki.

I'd be willing to do this as well if no one picks it up (I'd need to learn how to make a man page first).

(Also, I wasn't able to add a label to it for some reason)

### PR DESCRIPTION
**Pull request by MichaelAquilina**
Fixes #171. Adds a `scdoc` formatted files for generated man pages in Waybar.

The man pages are entirely based off the [Wiki](https://github.com/Alexays/Waybar/wiki/Configuration).

Test as follows (for example for waybar.5):

```
meson build
ninja -C build
man -l build/waybar.5 
```

Sub-modules names are formatted as `waybar-<module>`

### CODE REVIEW

**Comment by Alexays** [comment]
Thanks for this ❤️ , for a proper implemention, you can look at https://github.com/swaywm/sway/blob/master/meson.build#L95

**Comment by MichaelAquilina** [comment]
Added a new commit that adds building the man page to meson. Seems to work for me when I install it locally. Let me know how it goes for you!

**Comment by MichaelAquilina** [comment]
Based on the travis configuration, it seems like you need to build and upload the new Dockefile images for the build to pass?

**Comment by MichaelAquilina** [comment]
Separate question: How would you prefer to split man pages for modules within waybar? I was thinking of `man waybar-<module>`. For example, the battery module details would be found with `man waybar-battery`

**Comment by Alexays** [comment]
I think it would be cleaner to separate this in another folder with another meson file in this folder that compiles the documentation and in the main folder it includes the sub meson file and yeah `man waybar-battery` seems good

**Comment by MichaelAquilina** [comment]
Happy to go down that route if you believe it's better. Looking at the [subprojects documentation]( (https://mesonbuild.com/Subprojects.html) however makes me believe that this is more suited for actual libraries within the same project?

Maybe we can just move the `*.scd` files to a `man` folder and get the root `meson.build` script to look in that folder when generating files?

File: waybar-backlight.5.scd [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	typeof: string ++
+	Command to execute when performing a scroll up on the module.
+
+*on-scroll-down* ++
+	typeof: string
+	Command to execute when performing a scroll down on the module.
+
+*smooth-scrolling-threshold* ++
+	typeof: double
+	Threshold to be used when scrolling.
```
Comment (MichaelAquilina) [inline waybar-backlight.5.scd]: A tried a different format for representing the tables in the wiki here. I think it's easier to navigate than the tables generated by scdoc and scales better with narrow terminals (which in the case of using a tiling window manager is very important).

I alos find the syntax much easier to read than the table syntax used by scdoc.

Let me know what you think.

**Comment by MichaelAquilina** [comment]
Will wait for some more feedback before I continue adding more pages in case I need to change formatting or something

**Comment by Alexays** [comment]
Yeah move files into `/man` and i like the syntax you used, it's easy to write :)

**Comment by MichaelAquilina** [comment]
Sounds good :+1: was coicndentally working on this just now. 

With regards to the Docker images, would you mind updating dockerhub so that the images contain scdoc once this is merged?

**Comment by MichaelAquilina** [comment]
@Alexays I think this PR is ready to be reviewed and/or merged

**Comment by Alexays** [comment]
I'll update docker image once merged :)

File: meson.build [hunk 2]
```diff
@@ -153,6 +153,49 @@ install_data(
...
+        'waybar-sway-window.5.scd',
+        'waybar-sway-workspaces.5.scd',
+        'waybar-temperature.5.scd',
+        'waybar-tray.5.scd',
+    ]
+
+    foreach filename : man_files
+        topic = filename.split('.')[-3].split('/')[-1]
+        section = filename.split('.')[-2]
+        output = '@0@.@1@'.format(topic, section)
```
Comment (Alexays) [inline meson.build]: What the output for the `waybar.5.scd' file?

Comment (MichaelAquilina) [reply meson.build]: As in the path? If you generate the build with `meson build && ninja -C build` then the man pages will be in `build`.

Installing with `ninja -C build install` gives the following output though:

```
sudo ninja -C build install
[sudo] password for root:
ninja: Entering directory `build'
[0/1] Installing files.
Installing waybar to /usr/local/bin
Installing waybar.5 to /usr/local/share/man/man5
Installing waybar-backlight.5 to /usr/local/share/man/man5
Installing waybar-battery.5 to /usr/local/share/man/man5
Installing waybar-clock.5 to /usr/local/share/man/man5
Installing waybar-cpu.5 to /usr/local/share/man/man5
Installing waybar-custom.5 to /usr/local/share/man/man5
Installing waybar-idle-inhibitor.5 to /usr/local/share/man/man5
Installing waybar-memory.5 to /usr/local/share/man/man5
Installing waybar-mpd.5 to /usr/local/share/man/man5
Installing waybar-network.5 to /usr/local/share/man/man5
Installing waybar-pulseaudio.5 to /usr/local/share/man/man5
Installing waybar-sway-mode.5 to /usr/local/share/man/man5
Installing waybar-sway-window.5 to /usr/local/share/man/man5
Installing waybar-sway-workspaces.5 to /usr/local/share/man/man5
Installing waybar-temperature.5 to /usr/local/share/man/man5
Installing waybar-tray.5 to /usr/local/share/man/man5
Installing /home/michael/Development/waybar/./resources/config to /etc/xdg/waybar
Installing /home/michael/Development/waybar/./resources/style.css to /etc/xdg/waybar
```

**Comment by MichaelAquilina** [comment]
@Alexays what do you think of merging this PR if you are more or less happy with the approach and then we can all (collectively) work on improving things like typos, formatting, more docs etc. I already got a PR from someone else with suggested changes so I suspect this would be better of merged where everyone can contribute :)

**Comment by Alexays** [comment]
Yeah LGTM, Gonna merge it :)

**Comment by Alexays** [comment]
Thanks!

**Comment by Alexays** [comment]
Docker images is updating...

**Comment by MichaelAquilina** [comment]
Thanks @Alexays !

**Pull request merged**